ipa/verifier.go

```diff
@@ -9,14 +11,14 @@ import (
 // CheckIPAProof verifies an IPA proof for a committed polynomial in evaluation form.
 // It verifies that `proof` is a valid proof for the polynomial at the evaluation
 // point `evalPoint` with result `result`
-func CheckIPAProof(transcript *common.Transcript, ic *IPAConfig, commitment banderwagon.Element, proof IPAProof, evalPoint fr.Element, result fr.Element) bool {
+func CheckIPAProof(transcript *common.Transcript, ic *IPAConfig, commitment banderwagon.Element, proof IPAProof, evalPoint fr.Element, result fr.Element) (bool, error) {
 	transcript.DomainSep("ipa")
 
 	if len(proof.L) != len(proof.R) {
-		panic("L and R should be the same size")
+		return false, fmt.Errorf("vectors L and R should be the same size")
 	}
 	if len(proof.L) != int(ic.numRounds) {
-		panic("The number of points for L or R should be equal to the number of rounds")
+		return false, fmt.Errorf("the number of points for L and R should be equal to the number of rounds")
```

Comment:
I think this is fair, ie if we have a badly formatted proof, we can return false with an error. Probably in the future, we can just return errors because an error/false proof is not distinguishable from the clients perspectiv -- both of these require the client to not continue with any state transition they were about to do